test/plug_test.exs

```diff
@@ -1469,4 +1469,47 @@ defmodule PlugTest do
       assert load_and_authorize_resource(conn, opts) == expected
     end
   end
+
+  defmodule NonIdActions do
+    use ExUnit.Case, async: true
+
+    test "it throws an error when the non_id_actions is not a list" do
+      # when opts[:non_id_actions] is set but not as a list 
+      opts = [model: Post, non_id_actions: :other_action]
+      params = %{"id" => 1}
+      conn = conn(
+        %Plug.Conn{
+          private: %{phoenix_action: :other_action},
+          assigns: %{current_user: %User{id: 1}, authorized: true}
+        },
+        :get,
+        "/posts/other-action",
+        params
+      )
+
+      assert_raise Protocol.UndefinedError, "protocol Enumerable not implemented for :other_action", fn->
+        authorize_resource(conn, opts)
+      end
+    end
+
+    test "it authorizes the resource correctly when non_id_actions is a list" do
+      # when opts[:non_id_actions] is set as a list
+      opts = [model: Post, non_id_actions: [:other_action]]
+
+      params = %{"id" => 1}
+      conn = conn(
+        %Plug.Conn{
+          private: %{phoenix_action: :other_action},
+          assigns: %{current_user: %User{id: 1}, authorized: true}
+       },
+        :get,
+        "/posts/other-action",
+        params
+      )
+      expected = %{conn | assigns: Map.put(conn.assigns, :authorized, true)}
```

Comment:
I know the rest of the code doesn't use it yet, but could we use [Plug.Conn.assign](https://hexdocs.pm/plug/Plug.Conn.html#assign/3) here? e.g. `expected = Plug.Conn.assign(conn, :authorized, true)`. Looks a lot cleaner IMO :smile: